src/code.rs

```diff
@@ -422,25 +422,62 @@ impl {struct_name} {{
     ));
 
     buffer.push_str(&format!(r##"
-    /// Paginates through the table where page is a 0-based index (i.e. page 0 is the first page)
-    pub{async_keyword} fn paginate(db: &mut ConnectionType, page: i64, page_size: i64) -> QueryResult<PaginationResult<Self>> {{
+    /// Paginates through the table where page is a 1-based index (i.e. page 1 is the first page)
+    pub{async_keyword} fn paginate(db: &mut ConnectionType, param_page_starting_with_1: i64, param_page_size: i64, filter: {struct_name}Filter) -> QueryResult<PaginationResult<Self>> {{
```

Comment:
why update the parameter names with `param_`?

Reply:
it's in-line with the rest of the generated code, but I'll remove this :)

Reply:
to my understanding the style (implicitly) was to add the `param_` prefix to struct values as parameters, not separate options (like page size)